### ISSUE TITLE
bulkTranfer

### ISSUE THREAD

**Issue by frankbenoit** [usb4java/libusb4java#5]
Hi,

I think the LibUsb.c bulkTransfer(...) function is not implemented correctly.
It is using a ByteBuffer to receive the data to be transmitted. But it is not evaluating/updating the position/limit fields.
It only retrieves the array pointer and uses the capacity.
That forces the callee code to allocate the ByteBuffer for each call. It could have better performance if the starting to read at position, updates the read amount of bytes by increasing the position and take the limit into account (instead of the capacity).

This would make the "transferred" parameter no more needed.

Frank

**Comment by llongi** [comment]
Correct, this is by design. This libusb4java library is meant to map the C libusb API into Java, with no particular improvements for ease of use, just an as direct as possible port. So, as you would check transferred parameter in C, you also do so here. Higher level libraries can be build on top of this to make it easier to use, like the main usb4java code does. Updating the limit with the transferred parameter doesn't have any measurable impact on performance, it's the same as you'd have to do in the native code. Closing this, the API won't be changed as users rely on current behavior too.
